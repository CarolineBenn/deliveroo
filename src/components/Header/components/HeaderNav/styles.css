.mobile-header-nav {
  display: block;
  position: absolute;
  top: 45px;
  left: 0;
  background: #333;
  width: 100%;
  padding-bottom: 1rem;
}

@media (min-width: 678px) {
  .mobile-header-nav {
    display: none;
  }
}

.header-nav {
  display: none;
}

@media (min-width: 678px) {
  .header-nav {
    display: flex;
    flex-direction: row;
  }
}

@media (min-width: 678px) {
  .header-nav-menu {
    display: flex;
    flex-direction: row; 
  }
}

.header-nav-menu-item {
  padding: 1rem;
  display: block;
  color: #FFFFFF;
  text-decoration: none;
}
.header-nav-menu-item:hover,
.header-nav-menu-item:focus {
  color: #EEEEEE;
}

@media (min-width: 678px) {
  .header-nav-menu-item {
    display: inline-block;
  }
}