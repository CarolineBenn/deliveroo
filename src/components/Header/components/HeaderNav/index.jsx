import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const HeaderNav = ({
  expanded,
  mobile,
  navId,
}) => (
  <nav id={navId} className={`${mobile ? 'mobile-header-nav' : 'header-nav'}`} aria-expanded={expanded}>
    <ul className="unstyledList header-nav-menu">
      <li><a href="/" className="header-nav-menu-item">Home</a></li>
      <li><a href="/" className="header-nav-menu-item">Search</a></li>
      <li><a href="/" className="header-nav-menu-item">Â£8.55</a></li>
      <li><a href="/" className="header-nav-menu-item">Simon Rohrbach</a></li>
    </ul>
  </nav>
);

export default HeaderNav;

HeaderNav.propTypes = {
  /**
   * Bool to describe whether navigation is expanded or not (mobile)
   */
  expanded: PropTypes.bool,
  /**
   * Bool to describe whether component is being rendered on mobile or not.
   */
  mobile: PropTypes.bool,
  /**
   * String ID for the nav component.
   */
  navId: PropTypes.string,
};

HeaderNav.defaultProps = {
  expanded: false,
  mobile: false,
  navId: 'menuNav',
}