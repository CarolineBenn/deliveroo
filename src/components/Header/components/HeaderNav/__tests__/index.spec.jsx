import React from 'react';
import renderer from 'react-test-renderer';

import HeaderNav from '..';

test('HeaderNav renders correctly', () => {
  const tree = renderer
    .create(<HeaderNav />).toJSON();
  expect(tree).toMatchSnapshot();
});