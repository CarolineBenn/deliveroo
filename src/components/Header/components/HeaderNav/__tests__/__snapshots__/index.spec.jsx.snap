// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav renders correctly 1`] = `
<nav
  aria-expanded={false}
  className="header-nav"
  id="menuNav"
>
  <ul
    className="unstyledList header-nav-menu"
  >
    <li>
      <a
        className="header-nav-menu-item"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="header-nav-menu-item"
        href="/"
      >
        Search
      </a>
    </li>
    <li>
      <a
        className="header-nav-menu-item"
        href="/"
      >
        Â£8.55
      </a>
    </li>
    <li>
      <a
        className="header-nav-menu-item"
        href="/"
      >
        Simon Rohrbach
      </a>
    </li>
  </ul>
</nav>
`;
